### ISSUE TITLE
RDF to File very slow and memory intensive when saving as to JSON-LD

### ISSUE THREAD

**Issue by jakubklimek** [linkedpipes/etl#841]
In [this execution](https://etl.opendata.cz/#/pipelines/edit/canvas?pipeline=https:%2F%2Fetl.opendata.cz%2Fresources%2Fpipelines%2F1611299609194&execution=https:%2F%2Fetl.opendata.cz%2Fresources%2Fexecutions%2F1611299737651-785-5e23de6c-c913-45fc-94eb-0efe4b947d03) a 1GB Turtle file is to be saved as JSON-LD. The progress report hits 100% after a few minutes and then it takes 2 hours to produce the JSON-LD file, not to mention that this operation kills executor on a 32GB RAM machine. It finished on a 64GB machine.

**Comment by namedgraph** [comment]
Why on earth would you need JSON-LD that big? N-Triples and N-Quads are the most efficient formats for large RDF data.

**Comment by jakubklimek** [comment]
@namedgraph I agree. However, in the context of the Czech National Open Data Catalog, of which the RDF file is a dump, some developers prefer JSON over RDF. For them, we have a JSON schema specification, which turns the JSON into RDF using a certain JSON-LD context.

However, as we work with RDF internally, we also need to be able to generate the JSON dump of the catalog valid against that JSON schema. Also, we are building a GraphQL endpoint for the catalog, which works on top of a JSON dump.

This turtle file was to be used to create a JSON-LD dump, which would, in turn, be used as input to test out if [JSON-LD framing](https://www.w3.org/TR/json-ld11-framing/) can be used to create the JSON schema compliant JSON dump from the RDF dump automatically.
